service/controller/resource/namespaceauth/create.go

```diff
@@ -21,51 +22,60 @@ func (r *Resource) EnsureCreated(ctx context.Context, obj interface{}) error {
 		return microerror.Mask(err)
 	}
 
-	viewAccessWords := []string{"get", "list", "watch"}
-	viewAllRole, err := newViewAllRole(resources, viewAccessWords)
-	if err != nil {
-		return microerror.Mask(err)
+	viewAllRole.targetGroup = r.namespaceAuth.ViewAllTargetGroup
+	tenantAdminRole.targetGroup = r.namespaceAuth.TenantAdminTargetGroup
```

Comment:
Do we have to have globally defined variables and mutate them in some place? Can't we define them here instead?

Reply:
It is also in `delete.go` so I didn't want to duplicate it

Reply:
Replacing those vars with `newViewAllRole` funcs would work for. Otherwise I feel insecure :D

Reply:
Ok
I've moved vars into function as we actually don't need all of them in delete function